File: hz-back/src/main/java/com/beeasy/hzback/modules/system/entity/MessageRead.java
package com.beeasy.hzback.modules.system.entity;

import com.alibaba.fastjson.annotation.JSONField;
import com.beeasy.hzback.core.entity.AbstractBaseEntity;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@Entity
@Table(name = "t_message_read")
public class MessageRead extends AbstractBaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id = 0L;

//    用户ID(这里不需要使用关联)
    @JSONField(serialize = false)
    @ManyToOne
    User user;

//    Long fromId = 0L;
    Long toId = 0L;
//    Message.LinkType fromType = Message.LinkType.USER;
    Message.LinkType toType = Message.LinkType.USER;


    int unreadNum = 0;

    //会话ID, 不使用关联
//    @ManyToOne
//    MessageSession session;
}

File: hz-back/src/main/java/com/beeasy/hzback/modules/system/entity_kt/MessageRead.kt
package com.beeasy.hzback.modules.system.entity_kt

import com.alibaba.fastjson.annotation.JSONField
import com.beeasy.hzback.core.entity.AbstractBaseEntity
import com.beeasy.hzback.modules.system.entity.Message
import javax.persistence.*

@Entity
@Table(name = "t_message_read")
class MessageRead : AbstractBaseEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    //    用户ID(这里不需要使用关联)
    @JSONField(serialize = false)
    @ManyToOne
    lateinit var user: User

    //    Long fromId = 0L;
    var toId: Long? = null
    //    Message.LinkType fromType = Message.LinkType.USER;
    var toType: Message.LinkType = Message.LinkType.USER

    var unreadNum = 0

    //会话ID, 不使用关联
    //    @ManyToOne
    //    MessageSession session;
}